test/src/tests-session-manager.ts

```diff
@@ -0,0 +1,607 @@
+
+            const clearSessionTimeoutSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'clearSessionTimeout'
+            );
+            const setSessionTimerSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'setSessionTimer'
+            );
+
+            mpInstance._SessionManager.resetSessionTimer();
+            expect(clearSessionTimeoutSpy.called).to.equal(true);
+            expect(setSessionTimerSpy.called).to.equal(true);
+        });
+
+        it('should call startNewSessionIfNeeded', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+
+            const startNewSessionIfNeededSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'startNewSessionIfNeeded'
+            );
+
+            expect(startNewSessionIfNeededSpy.called).to.equal(false);
+
+            mpInstance._SessionManager.resetSessionTimer();
+            expect(startNewSessionIfNeededSpy.called).to.equal(true);
+        });
+    });
+
+    describe('#clearSessionTimer', () => {
+        it('should call clearTimeout', () => {
+            const clearTimeoutSpy = sinon.spy(window, 'clearTimeout');
+
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+
+            expect(clearTimeoutSpy.called).to.equal(false);
+            mpInstance._SessionManager.clearSessionTimeout();
+            expect(clearTimeoutSpy.called).to.equal(true);
+        });
+    });
+
+    describe('startNewSessionIfNeeded', () => {
+        it('should call startNewSession if sessionId is not available from Persistence', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+
+            // Session Manager relies on persistence check sid (Session ID)
+            sinon.stub(mpInstance._Persistence, 'getPersistence').returns({
+                gs: {
+                    sid: null,
+                },
+            });
+
+            mpInstance._Store.sessionId = null;
+
+            const startNewSessionSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'startNewSession'
+            );
+
+            mpInstance._SessionManager.startNewSessionIfNeeded();
+            expect(startNewSessionSpy.called).to.equal(true);
+        });
+
+        it('should NOT call startNewSession if sessionId is defined and Persistence is undefined', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+
+            // Session Manager relies on persistence check sid (Session ID)
+            // However, if persistence is undefined, this will not create a
+            // new session
+            sinon.stub(mpInstance._Persistence, 'getPersistence').returns(null);
+
+            mpInstance._Store.sessionId = null;
```

Comment:
`sessionId` is `null` here, but the test description says sessionId should be defined.  Should you set this to a string?

Reply:
There was a typo in the test suite label. it should have been `should NOT call startNewSession if sessionId is undefined and Persistence is undefined`

Reply:
Tiiiiny nit: I would say "not defined", since you are setting to null, which is not defined, and not `undefined`.  Or change the `null`s to `undefined` and then the test description is fine.